components/org.wso2.carbon.identity.application.authenticator.google/src/main/java/org/wso2/carbon/identity/application/authenticator/google/GoogleOAuth2Authenticator.java

```diff
@@ -128,21 +129,36 @@ protected String mapIdToken(AuthenticationContext context, HttpServletRequest re
      */
     private void validateCSRF(HttpServletRequest request, String clientID) throws AuthenticationFailedException {
 
+        boolean validCookies = false;
         boolean validateCSRF = true;
-        String enableCSRFValidationForGOT = getAuthenticatorConfig().getParameterMap()
-                .get(GoogleOAuth2AuthenticationConstant.ENABLE_CSRF_VALIDATION_FOR_GOT);
 
-        if (StringUtils.isNotBlank(enableCSRFValidationForGOT)) {
-            validateCSRF = Boolean.parseBoolean(enableCSRFValidationForGOT);
-        }
+        /*
+            This will skip the CSRF validation if it is done already in another layer i.e asgardeo-website.
+            That layer should handle the exception flow itself.
+         */
+        String validatedCSRF = request.getParameter(G_CSRF_VALIDATED);
+        if (StringUtils.isNotBlank(validatedCSRF)) {
+            String crossRefParamHalf = request.getParameter(G_CSRF_TOKEN);
+            if (StringUtils.isNotBlank(crossRefParamHalf) && validatedCSRF.equals(crossRefParamHalf)) {
+                validCookies = true;
+            }
+        } else {
+            String enableCSRFValidationForGOT = getAuthenticatorConfig().getParameterMap()
+                    .get(GoogleOAuth2AuthenticationConstant.ENABLE_CSRF_VALIDATION_FOR_GOT);
 
-        if (validateCSRF) {
-            boolean validCookies = validateCSRFCookies(request);
-            if (!validCookies) {
-                throw new AuthenticationFailedException(GoogleErrorConstants.ErrorMessages
-                        .CSRF_VALIDATION_FAILED_ERROR.getCode(), String.format(GoogleErrorConstants.ErrorMessages
-                        .CSRF_VALIDATION_FAILED_ERROR.getMessage(), clientID));
+            if (StringUtils.isNotBlank(enableCSRFValidationForGOT)) {
+                validateCSRF = Boolean.parseBoolean(enableCSRFValidationForGOT);
             }
+
+            if (validateCSRF) {
+                validCookies = validateCSRFCookies(request);
+            }
+        }
+
```

Comment:
remove this newline

Reply:
This is fixed 9754c47ccb2ecb12dec262ff6af5c04a6aea8e05